File: Swift/calc.swift
import Foundation

print("Voer getal 1 in: ")
let getal1 = Double(readLine() ?? "") ?? 0

print("Voer getal 2 in: ")
var getal2 = Double(readLine() ?? "") ?? 0

print("Voer de operand in: ")
let operand = readLine()

func bereken() -> Double {
    switch operand {
        case "+":
            return getal1 + getal2
        case "-":
            return getal1 - getal2
        case "*":
            return getal1 * getal2
        case "/":
            while getal2 == 0 {
                print("Delen door 0 is flauwekul, opnieuw: ")
                getal2 = Double(readLine() ?? "") ?? 0
            }
            return getal1 / getal2
        default:
            print("Slechter operand, => 0")
            return 0
        }
}

let resultaat = bereken()
print("\(getal1) \(operand ?? "") \(getal2) = \(resultaat)")

File: Ruby/calc.rb
while true
    print "Getal 1: "
    input = gets.chomp

    if (input == "stop")
        puts "Tot de volgende"
        break
    end

    getal1 = input.to_f

    print "Operator (+, -, *, /): "
    operator = gets.chomp

    print "Getal 2: "
    getal2 = gets.to_f

    resultaat = case operator
                when "+" then getal1 + getal2
                when "-" then getal1 - getal2
                when "*" then getal1 * getal2
                when "/" then 
                    while getal2 == 0
                    puts "Delen door 0 kan niet. Opnieuw: "
                    getal2 = gets.to_f
                    end
                    getal1 / getal2
                else
                    nil
                end

    if resultaat.is_a?(Numeric)
        puts "#{getal1} #{operator} #{getal2} = %.2f" % resultaat
    else
        puts "No bien operator"
    end
  
  puts "°" * 67 
end